### PR TITLE
fix: Panic reading multiple CSV files from cloud

### PR DESCRIPTION
**Pull request by nameexhaustion**
Fixes https://github.com/pola-rs/polars/issues/18053

### CODE REVIEW

File: crates/polars-io/src/file_cache/utils.rs [hunk 1]
```diff
@@ -95,7 +95,7 @@ pub fn init_entries_from_uri_list(
                         let cloud_path = object_path_from_str(&prefix)?;
 
                         let object_store =
-                            object_stores[std::cmp::min(i, object_stores.len())].clone();
+                            object_stores[std::cmp::min(i, object_stores.len() - 1)].clone();
```
Comment (nameexhaustion) [inline crates/polars-io/src/file_cache/utils.rs:98]: silly mistake 😅

we unfortunately can't add this to our test suite because it needs a real bucket

Comment (mghen) [reply crates/polars-io/src/file_cache/utils.rs:98]: Happens to the best of us 😅

Comment (mghen) [reply crates/polars-io/src/file_cache/utils.rs:98]: Thanks for the quick fix!

Comment (thomasfrederikhoeck) [reply crates/polars-io/src/file_cache/utils.rs:98]: @nameexhaustion  Would it make sense to add a test which uses Azurite (using docker) or similar to test reading from "the cloud"?  https://github.com/Azure/Azurite

**Pull request merged**